src/useGetPosts/index.js

```diff
@@ -66,24 +66,22 @@ import { apiClient } from '../utils';
  */
 function useGetPosts( args = {}, clientId, postType ) {
 	const [ options, setOptions ] = useState( [] );
-	const [ query, setQuery ] = useState( '' );
-	const [ loading, setLoading ] = useToggle();
+	const [ query, setQuery ] = useState( [] );
```

Comment:
Initializing query state with an empty string is also done by design to be notified the moment that component is mounted and is not yet established any REST connection to retrieve the array of posts object. 

In the event that the API fails to retrieve expected posts results from the database, an empty array is returned to ensure that hook is moved on to the catch method instead.

Reply:
> Initializing query state with an empty string is also done by design to be notified the moment that component is mounted and is not yet established any REST connection to retrieve the array of posts object.

Would you be able to elaborate on this point? Is this value being used anywhere where it is checked against being an empty string?

Reply:
The initial placeholder value could be helpful to identify the moment where the component is mounted but yet to be entered into the hook to establish the API connection. We can expect an array value returned only when the connection is established regardless of the status result.

Reply:
> Is this value being used anywhere where it is checked against being an empty string?

Yes. Several blocks are still checking for the data type to render the loading indicator in place of the actual data during the API connection. [Example](https://github.com/sixach/wp-block-testimonials/blob/main/src/Edit.js#L87).

Reply:
@kuserich I was planning to publish a new release for this package and wondered whether we could revert this particular commit back?

Reply:
Yes, I have reset the default value for query to `''` in [this commit](https://github.com/sixach/wp-react-hooks/pull/55/commits/c67ff98f8f4ee0f78b6c0c0f3584d2e3dae496f7).

Reply:
Thanks for elaborating, @mahdiyazdani . In the example that you have shared, we are essentially checking against the data type to check whether the data is still being loaded. This works because the type of the default value changes regardless of the result of the query. However, by doing so, I believe we are relying on logic rather than data and I would find an additional boolean parameter more appropriate to distinguish this type of result. Particularly, I would propose updating the return value with an additional value `hasLoaded` (or `hasLoadedOnce`) which holds `true` whenever the query was run at least once. What do you think? If you agree or if you would like to see an example to play with, I would be able to update this PR or create a new proposal in a new PR.

Reply:
Absolutely. I also find it practical to have a variable exported that states the hook is already tried to establish the connection at least once. However, where would be the best place to introduce this conditional statement and not check aginst every time dependencies of the hook receive an update?

Reply:
In the example, `isLoading` would be sufficient to render the loading spinner while the data is being queried. The only difference is that the loading spinner is also displayed when the query is changed subsequently. This could be considered the appropriate behavior given that we are indeed querying data (and not showing the loading spinner would be optional IMO).

Either way, the original behavior can be restored my switching the order of the conditions:

```JS
function Edit( { attributes, clientId, setAttributes } ) {
    const { category, ids, limit, order, orderby } = attributes;
    const postType = applyFilters( 'sixa.testimonialsPostType', 'testimonial-item' );
    const taxonomy = applyFilters( 'sixa.testimonialsTaxonomy', 'testimonial-item-category' );
    const blockProps = useBlockProps();
    const { isLoading, postsOptions, postsQuery } = useGetPosts( { [ taxonomy ]: category || ' ', order, orderby }, clientId, postType );
    const { termsOptions } = useGetTerms( taxonomy );
    const { havePosts, maxLimit, slicedQuery } = usePreparePosts( ids, limit, postsQuery );
    const className = blockClassName( blockProps?.className );

    return (
        <div { ...blockProps }>
            {
                havePosts ? (
                    <Loop className={ className } query={ slicedQuery } />
                ) : isLoading ? (
                    <Loading />
                ) : (
                    <Notice css={ { margin: 0 } } isDismissible={ false } status="warning">
                        { __( 'No slides found to display.', 'sixa-block-testimonials' ) }
                    </Notice>
                )
            }
            <Inspector
                attributes={ attributes }
                maxLimit={ maxLimit }
                postsOptions={ postsOptions }
                setAttributes={ setAttributes }
                termsOptions={ termsOptions }
            />
        </div>
    );
}
```

Generally speaking, I would assume that `hasLoadedOnce` would only be required if a component requires to distinguish between an empty result set and no result set after loading. Thus, this boolean would be checked only in a small number of rare instances. In the example, this would be in the last block where two different Notices (or whatever is more appropriate) could be displayed.